noir-projects/aztec-nr/uint-note/src/uint_note.nr

```diff
@@ -233,12 +235,44 @@ impl PartialUintNote {
         context.push_note_hash(self.compute_complete_note_hash(value));
     }
 
-    fn compute_note_completion_log(self, value: u128) -> [Field; 2] {
+    /// Completes the partial note, creating a new note that can be used like any other UintNote.
+    pub fn complete_from_private(self, value: u128, context: &mut PrivateContext) {
+        // A note with a value of zero is valid, but we cannot currently complete a partial note with such a value
+        // because this will result in the completion log having its last field set to 0. Public logs currently do not
+        // track their length, and so trailing zeros are simply trimmed. This results in the completion log missing its
+        // last field (the value), and note discovery failing.
+        // TODO(#11636): remove this
+        assert(value != 0, "Cannot complete a PartialUintNote with a value of 0");
+
+        // We need to do two things:
+        //  - emit an unencrypted log containing the public fields (the value) via the private log channel. The
+        //  contract will later find it by searching for the expected tag (which is simply the partial note
+        //  commitment).
+        //  - insert the completion note hash (i.e. the hash of the note) into the note hash tree. This is typically
+        //  only done in private to hide the preimage of the hash that is inserted, but completed partial notes are
+        //  inserted in public as the public values are provided and the note hash computed.
+        context.emit_private_log(
+            self.compute_note_completion_log_padded_for_private_log(value),
+            NOTE_COMPLETION_LOG_LENGTH,
+        );
```

Comment:
I find the `emit_private_log` API on `context` quite weird as we kinda pass a `BoundedVec` here but not really. Shall we just change it to use `BoundedVec`? Then I would not have to do the ugly manual padding here.